### DISCUSSION TITLE
Support for CalDAV

### DISCUSSION THREAD

**Discussion by TheZoker** [chrisvel/tududi#246] [Feature requests] [answered]
Would be amazing if this app supports caldav synchronisation. This would allow to use this as a webfrontend for a caldav server (e.g. etesync or Baikal etc.) and to use it with the amazing tasks.org android/iOS app. 

I currently use the app and I'm missing a nice web interface to also work on my tasks. Tu Du Di looks amazing and would perfectly fit that requirement (if a docker image is available).

Would appriciate to hear some feedback on this idea! 

Bonus point: You instantly have a app to work alongside your project :D

**Comment by skyrunner15** [comment]
+1 so I can use it on my Android phone like DavX+Tasks.org

**Comment by chrisvel** [comment]
@TheZoker Can you please explain in a sentence how could this feature help with personal task management ? The reason I am asking is because I have no experience with CalDAV and I want to evaluate If this would be helpful for me and for other people too.

**Comment by TheZoker** [comment]
@chrisvel So the main idea would be, to be able to manage task from the tududi and from a mobile app. And since CalDAV is a open standard, there are already many apps that support this (e.g. tasks.org, DavX etc.)
With a CalDAV integration I can add tasks from my mobile app and they will popup in the webapp. When I mark a task as done in the webapp it will also be marked as done in the mobile app. And so on. 
The main benefit is, that I can manage the tasks no matter which device I use. I can add tasks on the go and check them when I'm working on my notebook.
Does this make sense? Let me know if you need more information! 😄

**Comment by chrisvel** [comment]
@TheZoker yeah, I know in generic terms what CalDAV is and I have used DavX in the past but I haven't developed anything related to it ever. I guess it could be easy to add an integration - the difficult part would be to make it compatible with the models and the schema. I promise to research it though and come up in a few days with some valuable results. Thanks a lot!

**Reply by dev-inside** [reply]
Maybe tsdav could help with the heavy lifting :) https://tsdav.vercel.app/

**Comment by groetg** [comment]
+1 for me too.

**Comment by xlarin** [comment]
Interesting feature, I would like to see it in the app.

**Comment by bytefrostdev** [comment]
+1 for CalDAV 🗓️

**Comment by meichthys** [comment]
+1 this would allow integration with a host of other apps and services.

**Comment by CiprianStoicescu** [comment]
+1

**Comment by Den4ik** [comment]
+1

**Reply by thekiwismarthome** [reply]
+1 for as well please

**Comment by Cyber1000** [comment]
I support that as well, just an addition to discuss - there may be two options:
- as the main idea here we speak about a caldav-client-integration:
   - pro: if you already have a caldav-server you can use this in your preferred way
   - con: maybe tududis models don't fit or aren't easy to map between caldav-server and tududi, although I would expect most parts to be consistent. The biggest problem I see may be reccurence (and here things like "Repeat after completion". A count of occurences additional to and end date may also be neccessary to be equal to rfc 5545)
   - way to go: integrate https://tsdav.vercel.app/ as @dev-inside mentioned
 - imho another way may be to discuss - a caldav-server-integration:
    - I have seen this in another similar project (https://vikunja.io/docs/caldav/) in an early stage and for me it looks quite interesting
    - pro: full control, of your models, you can exactly control what your caldav-clients get
    - con: maybe some workflows won't work anymore (if you currently rely on another integrated caldav-server and can't use a caldav-client there)
    - way to go: integrate https://github.com/forwardemail/caldav-adapter

For now I'm not entirely sure which is better. In both ways we would also need to discuss how we solve the problem if a task is changed on both sides.

**Comment by chrisvel** [comment]
I am picking up features to add to v1.1.0. Can you please post any requirements, expectations, use cases?

**Comment by fajremvp** [comment]
+1

**Comment by chrisvel** [comment, marked as answer]
CalDav support has been added on `v.1.1.0`. 

Please open bug tickets and new feature requests for any item related to this feature. 

Thanks!
